File: Data Structures & Algorithms/valid-sudoku/submission-1.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        r = {}
        c = {}
        g = {}

        for x in range(len(board)):
            for y in range(len(board[x])):
                curr = board[x][y]
                if curr == ".":
                    continue
                if x not in r:
                    r[x] = set()
                if y not in c:
                    c[y] = set()
                if (x//3, y//3) not in g:
                    g[(x//3, y//3)] = set()
                if curr in r[x] or curr in c[y] or curr in g[(x//3, y//3)]:
                    return False
                r[x].add(curr)
                c[y].add(curr)
                g[(x//3,y//3)].add(curr)
        return True

File: Data Structures & Algorithms/reorder-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {
        ListNode fast = head;
        ListNode slow = head;
        while (fast != null && fast.next != null) {
            fast = fast.next.next;
            slow = slow.next;
        }

        ListNode curr = slow;
        ListNode temp = null;
        ListNode prev = null;
        while (curr != null) {
            temp = curr.next;
            curr.next = prev;
            prev = curr;
            curr = temp;
        }
        ListNode first = head;
        ListNode second = prev;
        while (second.next != null) {
            ListNode t1 = first.next;
            ListNode t2 = second.next;
            first.next = second;
            second.next = t1;
            first = t1;
            second = t2;
        }
    }
}
